src/dpp/webhook.cpp

```diff
@@ -33,6 +33,25 @@ webhook::webhook() : managed(), type(w_incoming), guild_id(0), channel_id(0), us
 {
 }
 
+webhook::webhook(std::string webhook_url)
+{
+	token = webhook_url.substr(webhook_url.find_last_of("/") + 1);
+	try
+	{
+		id = std::stoull(webhook_url.substr(sizeof("https://discord.com/api/webhooks/") - 1, webhook_url.find_last_of("/")));
+	}
+	catch (std::exception& e)
+	{
+		throw dpp::logic_exception(std::string("Your webhook's URL failed to parse! ID invalid.\n") + e.what());
+	}
+}
+
+webhook::webhook(snowflake webhook_id, std::string webhook_token)
```

Comment:
string parameters should be `const std::string&`

Reply:
the ID can be made a constant too, right?

Reply:
Yes.